### PR TITLE
[PKG-5097] Update to 0.30.1

### PR DESCRIPTION
**Pull request by ViridianMelody**
modin 0.30.1

**Destination channel:** defaults

### Links

- [PKG-5097] 
- [Upstream repository](https://github.com/modin-project/modin/tree/0.30.1)
- [Upstream changelog/diff](https://github.com/modin-project/modin/compare/0.28.1...0.30.1)

### Explanation of changes:

- Adjust test commands for relocated test files.
- Add `async-timeout` to `modin-ray`. 
- Update dependency pinnings.
- Confirmed that `versioneer` patch is still necessary for python 3.12

[PKG-5097]: https://anaconda.atlassian.net/browse/PKG-5097?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by lorepirri** [comment]
@ViridianMelody you are right https://github.com/ray-project/ray/issues/41267 and probably our Ray needs a fix (see comments in the thread)

File: recipe/meta.yaml [hunk 1]
```diff
@@ -143,6 +144,7 @@ outputs:
         - {{ pin_subpackage('modin-core', exact=True) }}
         - ray-default >=1.13.0,!=2.5.0
```
Comment (danpetry) [inline recipe/meta.yaml:145]:
```suggestion
        - ray-default >=2.1.0,!=2.5.0
```

File: recipe/meta.yaml [hunk 2]
```diff
@@ -143,6 +144,7 @@ outputs:
...
         - pyarrow >=7.0.0
```
Comment (danpetry) [inline recipe/meta.yaml:146]:
```suggestion
        - pyarrow >=10.0.1
```

File: recipe/meta.yaml [hunk 3]
```diff
@@ -143,6 +144,7 @@ outputs:
...
+        - async-timeout
```
Comment (danpetry) [inline recipe/meta.yaml:147]: where's this coming from?

**Comment by danpetry** [comment]
would you be able to add a comment to remove async-timeout when ray is updated or whatever the best action would be?

File: recipe/meta.yaml (hunk 3 above)
Comment (ViridianMelody) [reply recipe/meta.yaml:147]: Basically, it's a transitive dependency of `ray-core` that's no longer being pulled in by `aiohttp`. I'm working on a repodata hotfix for it.

**Review by lorepirri — APPROVED** [review]

File: recipe/meta.yaml [hunk 4]
```diff
@@ -79,8 +79,8 @@ outputs:
       run:
         - python
         - packaging >=21.0
-        - pandas ==2.2.1
-        - numpy >=1.22.4
+        - pandas >=2.2,<2.3
+        - numpy >=1.22.4,<2
         - fsspec >=2022.11.0
         - psutil >=5.8.0
```
Comment (skupr-anaconda) [inline recipe/meta.yaml:85]: I think we need `run_constrained` for `numexpr<2.8.5`, see https://github.com/modin-project/modin/blob/0.30.1/requirements-dev.txt#L35

Comment (ViridianMelody) [reply recipe/meta.yaml:85]: I believe `numexpr !=2.8.5` would be okay, judging by this PR: https://github.com/modin-project/modin/pull/7296.

**Pull request merged**